docs/migrating.md

```diff
@@ -0,0 +1,301 @@
+native imports, pending an upgrade of GlimmerX's own dependencies.
+
+### Background
+
+Prior to version `0.7.4`, Glint required users to import the factories and base classes for components, helpers and
+modifiers from custom environment-specific paths. This was because the "native" versions of those entities couldn't
+capture enough information for Glint to typecheck them in a template with reasonable fidelity.
+
+Consider `@glimmer/component` as an example.
+
+```typescript
+import Component from '@glimmer/component';
+
+export interface MyComponentArgs {
+  message: string;
+}
+
+export default class MyComponent extends Component<MyComponentArgs> {
+  // ...
+}
+```
+
+Knowing only about the `MyComponent`'s expected `@arg` values, Glint couldn't provide any validation or assistance
+for what blocks the component accepts, what parameters it yields to those blocks, or what kind of modifiers would
+be valid to apply to it.
+
+### Signatures
+
+Ember RFC [#748](https://github.com/emberjs/rfcs/pull/748) formalized a notion of "component signatures" based on
+exploratory work Glint had done to find ways of expression the type information that `Args` alone couldn't capture.
+
+The `@glimmer/component` package, as well as `ember-modifier` and `@types/ember__component` have been updated based on
+the results of that RFC, and starting with version `0.7.4` of Glint, users may opt into using the regular import paths
+and standardized signature formats those packages have adopted.
+
+### Opting into Native Integration
+
+In order to opt into Glint's augmentation of the native component, modifier and helper imports, add the following import
+somewhere in your project (likely below wherever you have `import '@glint/environment-ember-loose'`):
+
+```typescript
+import '@glint/environment-ember-loose/native-integration';
+```
+
+In version `0.8.0`, integration will be enabled by default and this extra import will be unnecessary.
+
+### New Signature Formats
+
+As you move each component, modifier and helper in your project from the `environment-ember-loose` import path to the
+native one, you'll also need to update its signature to the standardized format. You can make this migration
+incrementally or all at once, depending on the size of your project.
+
+#### Components
+
+For components, there are two key changes (see [the relevant section](https://github.com/dfreeman/ember-rfcs/blob/glimmer-component-signature/text/0748-glimmer-component-signature.md#invokablecomponentsignature) in the RFC):
+
+- `Yields` has become `Blocks`. This key has a more complex notional desugaring, but the shorthand is compatible with
+  how `Yields` worked before.
+- `Args` and `PositionalArgs` have been merged into `Args: { Named: ...; Positional: ... }`. If your component only has
+  named args, the wrapping layer can be skipped and you can continue to use `Args: MyNamedArgs` as before.
```

Comment:
Maybe adding a note here that this only particular affects Classic Component users, since it's the same for Glimmer Component users?